app/src/main/java/ar/com/wolox/android/cookbook/scanqr/ScanQrActivity.kt

```diff
@@ -0,0 +1,78 @@
+package ar.com.wolox.android.cookbook.scanqr
+
+import android.app.Activity
+import android.content.Intent
+import android.widget.Toast
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.scanqr.scanerror.ScanErrorFragment
+import ar.com.wolox.android.cookbook.scanqr.scanmenu.ScanMenuFragment
+import ar.com.wolox.android.cookbook.scanqr.scansuccess.ScanSuccessFragment
+import ar.com.wolox.wolmo.core.activity.WolmoActivity
+import com.google.zxing.integration.android.IntentIntegrator
+import javax.inject.Inject
+
+class ScanQrActivity : WolmoActivity(), ScanQrView {
+
+    @Inject
+    lateinit var presenter: ScanQrPresenter
+    var result: String? = null
+
+    override fun init() {
+        presenter = ScanQrPresenter().apply {
```

Comment:
Well, I'm not getting why are you setting the presenter like this. There was not posibility to use a Fragment that calls a Presenter?